mailarchive/mailarchive-subetha/src/java/org/sakaiproject/mailarchive/SakaiMessageHandlerFactory.java

```diff
@@ -172,8 +177,12 @@ public void from(String from) throws RejectException {
             public void recipient(String to) throws RejectException {
                 SplitEmailAddress address = SplitEmailAddress.parse(to);
 
-                if (serverConfigurationService.getServerName().equalsIgnoreCase(address.getDomain())) {
-                    // || serverConfigurationService.getServerNameAliases().contains(address.getDomain())) {
+                if(to.startsWith(serverConfigurationService.getString("msgcntr.messages.header.from.reply"))) {
+                    isMessageId = true;
+                    String id = to.replace(serverConfigurationService.getString("msgcntr.messages.header.from.reply"), "").split("@")[0];
```

Comment:
Create a constant for the string "msgcntr.messages.header.from.reply" and use StringUtils.EMPTY instead of ""